unicode-data/exe/Parser/Text.hs

```diff
@@ -256,6 +267,296 @@ isHangul c = n >= hangulFirst && n <= hangulLast
 -- Parsers
 -------------------------------------------------------------------------------
 
+mkHaskellConstructor :: String -> String
+mkHaskellConstructor = fmap $ \case
+    '-' -> '_'
+    ' ' -> '_'
+    c   -> if isAscii c && isAlphaNum c
+        then c
+        else error ("Unsupported character: " <> show c)
+
+genBlocksModule
+    :: Monad m
+    => String
+    -> Fold m BlockLine String
+genBlocksModule moduleName = done <$> Fold.foldl' step initial
+    where
+
+    done (blocks, defs, ranges) = let ranges' = reverse ranges in unlines
```

Comment:
Can probably use `Streamly.Internal.Unicode.String.str`.

Reply:
I find reference to neither the module nor the function on Hackage.